merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -531,6 +531,16 @@ public static SimulationResults computeResults(
             activityChildren.getOrDefault(activityId, Collections.emptyList()),
             (activityParents.containsKey(activityId)) ? Optional.empty() : Optional.of(activityId)
         ));
+      } else if (state instanceof ExecutionState.AwaitingChildren<?> e){
+        final var inputAttributes = taskInfo.input().get(task.id());
+        unfinishedActivities.put(activityId, new UnfinishedActivity(
+            inputAttributes.getTypeName(),
+            inputAttributes.getArguments(),
+            startTime.plus(e.startOffset().in(Duration.MICROSECONDS), ChronoUnit.MICROS),
+            activityParents.get(activityId),
+            activityChildren.getOrDefault(activityId, Collections.emptyList()),
+            (activityParents.containsKey(activityId)) ? Optional.empty() : Optional.of(activityId)
+        ));
```

Comment:
Now that we're exhaustively checking for each of the possible `ExecutionState` types it may be good to add an `else` here to guarantee every subtype is being handled when testing:
```java
    } else {
      throw new Error("Unexpected subtype of %s: %s".formatted(ExecutionState.class, state.getClass()));
    }
```
When that JEP for pattern matching on sealed types is available this can be taken out.

Reply:
Good point! I've added this.